### DISCUSSION TITLE
RFC: Single `tamagui` package

### DISCUSSION THREAD

**Discussion by nandorojo** [tamagui/tamagui#330] [Ideas]
What if all packages were lumped into a single `tamagui`? All subpackages could be imported via `tamagui/lucide-icons`, etc. This might be a bad idea. But I struggle to see what's wrong with it other than the fact that your node modules gets bigger. Tree shaking shouldn't get affected. This is what `next`, `moti`, etc. other libraries do. 

Thought I'd toss it out there. While maintaining libraries with separate packages, I learned that versioning can get super difficult for consumers, especially when the packages import each other and rely on context. 

It won't affect me that much since I have experience with monorepos etc., but I know it's tough to manage, especially for beginners.

**Comment by dohomi** [comment]
@nandorojo isn't that already done via https://github.com/tamagui/tamagui/tree/master/packages/tamagui?

**Reply by natew** [reply]
Yea mostly is there, I think it has basically every package except the icons and fonts (design system specific stuff, which is all grouped under `config-base`).

**Reply by nandorojo** [reply]
i would also consider removing packages like gradient from the main export. could have side effects for users who don’t install it. that’s why `moti/skeleton` is a useful import
